modules/codepipeline/main.tf

```diff
@@ -0,0 +1,69 @@
+## TODO: DeletionPolicy = Retain
+resource "aws_s3_bucket" "codepipeline" {
+  bucket = "artifact_store"
+  versioning {
+    enabled = true
+  }
+}
+
+resource "aws_codepipeline" "hyku" {
+  name     = "hyku-codepipeline"
+  role_arn = "${aws_iam_role.codepipeline.arn}"
+
+  artifact_store {
+    location = "${aws_s3_bucket.codepipeline.bucket}"
+    type     = "S3"
+  }
+
+  stage {
+    name = "Source"
+
+    action {
+      name             = "SourceAction"
+      category         = "Source"
+      owner            = "AWS"
+      provider         = "S3"
+      version          = "1"
+      output_artifacts = ["SourceOutput"]
+
+      configuration {
+        S3Bucket    = "${var.S3Bucket}"
+        S3ObjectKey = "${var.S3Key}"
+      }
+    }
+  }
```

Comment:
An upgrade to this work would be to have it deploy straight from a GitHub repo rather than deploying from an intermediate S3 bucket

Reply:
Right now I do have a strong preference towards doing a direct port where we can help it, before doing changes to how things work.  I'm good to do as a follow-on if we have time but just generally there's enough moving pieces in all of this that it's pretty complicated.  The individual modules are minor to not too bad, but when we start to wire it all together I think having an almost-identical model to compare to is going to be really useful.

There's also the thing where I'm personally not that experienced with various AWS bits besides the higher level EC2 and a few others, and keeping both conversion and changes in my head does worry me there.  :P

Reply:
ok, fair enough. Would you mind making a ticket to remind us to come back around to this?